### ISSUE TITLE
[Feature] Integration with Botania to allow spark-based mana transfer across entangled blocks.

### ISSUE THREAD

**Issue by dylithian** [SuperMartijn642/Entangled1.12#58]
**Is your feature request related to a problem? Please describe.**
Consider the following scenario:
![image](https://github.com/SuperMartijn642/Entangled/assets/64990317/e46722ad-a00e-4688-a056-dd85038a6336)
Here I have a mana pool with a spark on it. If I place a recessive augment on the spark, it can be used to give mana to certain other mana pools within range.

Then:
![image](https://github.com/SuperMartijn642/Entangled/assets/64990317/e7d8ced5-229e-4634-b765-0ff77d62b792)
I place an entangled block some arbitrary distance away, and I entangle it to the mana pool from above.

If I place another mana pool directly adjacent to the mana pool, nothing happens. As far as Botania is concerned, the entangled block is not actually a Mana Pool with a spark on it, but just a random block from the Entangled mod.

(Example setup):
![image](https://github.com/SuperMartijn642/Entangled/assets/64990317/75854cf7-258d-41fb-b470-de2895cf21bf)

**Describe the solution you'd like**
If the idea is that entangled blocks "are" whatever they're entangled to, at least one of two things should be possible, but as of time of writing, currently are not.

1) Botania mana pools (non-entangled) should treat the entangled block like another mana pool if the entangled block's target has a spark, and be able to give or receive mana from the entangled block's target pool. Basically, they should treat the entangled block like a mana pool with the entangled target's mana and spark configuration.

The configuration for this is basically,

Real mana pool -> Entangled block targeting real mana pool -> Entangled Block -> Mana pools distant from the real mana pool but with the mod's own radius of the entangled block

In this version, mana gets transferred by the distant mana pools, where they give or receive mana from the entangled block's target as though it were there.

2) Entangled blocks simply behave like mana pools if configured to target mana pools, and they synchronize their mana capacity somehow with the entangled target. This would be implemented as being able to put a spark on an entangled block if the entangled target is a mana pool, allowing that entangled block to transfer mana between its entangled target and other mana pools within normal range of the entangled block.

In this version, mana gets transferred by the entangled block itself, and it simply gives or takes mana based on comparing its internal mana pool to its target to make sure their mana values are the same at all times. The player could then place a spark on the entangled block to "interact" with this fake internal mana pool, manifesting as a spark on the entangled block.

Basically, tldr; Entangled Blocks should allow me to use sparks to bypass Botania's in-built spark distance radius.

**Comment by KamiN0Baka** [comment]
I feel like this is because of the way sparks work. They don't actually modify the mana pool itself, the spark in your example case is a separate entity connected to the mana pool that pulls mana out of the pool and transfers it to other mana pools near the spark, not the mana pool the spark is on. You'd need a way to entangle the spark itself for it to work right. Either that or allow an entangled block linked to a mana pool to accept sparks, but that seems like it might be harder to do.

Edit: I just realized how old this feature request is, but my thoughts on it still stand.
